platform/darwin/src/MGLBackgroundStyleLayer.mm

```diff
@@ -51,6 +51,9 @@ - (void)addToMapView:(MGLMapView *)mapView belowLayer:(MGLStyleLayer *)otherLaye
 - (void)removeFromMapView:(MGLMapView *)mapView
 {
     auto removedLayer = mapView.mbglMap->removeLayer(self.identifier.UTF8String);
+    if (!removedLayer) {
+        return;
+    }
     _pendingLayer = std::move(reinterpret_cast<std::unique_ptr<mbgl::style::BackgroundLayer> &>(removedLayer));
```

Comment:
Not part of this PR, but this cast is undefined behavior. You need to write something like:

```
template <class T, class U>
std::unique_ptr<T> dynamic_ptr_cast(std::unique_ptr<U> p) {
    T* t = dynamic_cast<T*>(p.get());
    if (!t) return nullptr;
    p.release();
    return t;
}
```

And be aware that this will return a null `unique_ptr` (and deallocate the input) if the `dynamic_cast` fails, i.e. if the input is not the anticipated derived type. In this particular code, it looks like this could happen if the user was aliasing `*StyleLayer` objects while at the same time removing a layer and adding another layer with the same identifier but a different layer type.